Lib/test/test_dataclasses/__init__.py

```diff
@@ -1524,6 +1524,24 @@ class A(types.GenericAlias):
         self.assertTrue(is_dataclass(type(a)))
         self.assertTrue(is_dataclass(a))
 
+    def test_is_dataclass_for_subclass(self):
+        # Test to ensure that inheritance from a dataclass works correctly
+        # and that the subclass is also recognized as a dataclass.
+        @dataclass
+        class X:
+            y: int
+
+        class Z(X):
+            z: int = 0  # Adding a default field to demonstrate inheritance and dataclass behavior
+
+        # Check if X is a dataclass
+        self.assertTrue(is_dataclass(X), "X should be a dataclass")
+        # Check if Z is a dataclass, it should inherit the dataclass behavior from X
+        self.assertTrue(is_dataclass(Z), "Z should be a dataclass because it inherits from X")
+        # Create an instance of Z and check if the default values are set correctly
+        z_instance = Z(y=5)
+        self.assertEqual(z_instance.y, 5, "The value of y should be set to 5")
+        self.assertEqual(z_instance.z, 0, "The default value of z should be 0")
```

Comment:
These checks are not related to `is_dataclass` so don't belong in this test.

Reply:
Removed unnecessary piece of code !